include/openssl/cms.h

```diff
@@ -152,6 +152,7 @@ int CMS_decrypt_set1_password(CMS_ContentInfo *cms,
 STACK_OF(CMS_RecipientInfo) *CMS_get0_RecipientInfos(CMS_ContentInfo *cms);
 int CMS_RecipientInfo_type(CMS_RecipientInfo *ri);
 EVP_PKEY_CTX *CMS_RecipientInfo_get0_pkey_ctx(CMS_RecipientInfo *ri);
+CMS_ContentInfo *CMS_AuthEnvelopedData_create(const EVP_CIPHER *cipher);
```

Comment:
This function needs documenting

Reply:
I agree. The only thing is that there is currently no documentation for `CMS_EnvelopedData_create` which should be probably also present. Do you think that it would make more sense if I created a new PR and added some docs for `CMS_EnvelopedData_create` and possibly some other CMS related functions? Then `CMS_AuthEnvelopedData_create` specific part would be added in this PR.

Reply:
I'd be delighted if you opened a new PR and documented some of the other functions in another PR. Strictly all I need to approve this PR is the ensure that this new function is documented (whether or not the others are). I'll let you decide how best to tackle that.

Reply:
Added docs for `CMS_EnvelopedData_create` in #11529 . Will add the auth part into it once merged.

Reply:
Docs added.